### PR TITLE
Totem Edit UI

### PR DESCRIPTION
**Pull request by mouhb**
First version of the totem Edit UI
In this PR I make the following modifications
- Put Totem edit actions in popups (Edit sections and items)
- Group the macros of the TotemCode.TotemSheet document in a new document TotemCode.TotemMacros, to be reused.
- Make a little modifications to the macros to adpat theme to the popup display mode

### CODE REVIEW

**Comment by fveron** [comment]
By doing a very quick reading i saw two things.
- The indentation is not regular, should we adjust this ?
- use of "console.log", i my mind this method is not working on every browser

File: src/main/resources/TotemCode/SimpleTotemItemSheet.xml [hunk 1]
```diff
@@ -12,8 +12,8 @@
   <customClass/>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
   <creationDate>1398855257000</creationDate>
-  <date>1422027885000</date>
```
Comment (ldubost) [inline src/main/resources/TotemCode/SimpleTotemItemSheet.xml:15]: We should avoid date changes. The commit should normally use the same date as before. This is not a blocker so it can be commited as is but it would be better to avoid this next time.

File: src/main/resources/TotemCode/TotemMacros.xml [hunk 2]
```diff
@@ -0,0 +1,275 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ldubost) [inline src/main/resources/TotemCode/TotemMacros.xml:1]: This commit would have been better as a separate commit. This is not a blocker so it can be commited as is but it would be better to avoid this next time.

File: src/main/resources/TotemCode/TotemSheet.xml [hunk 3]
```diff
@@ -108,8 +108,8 @@
   initialize: function() {
     this.hookEditOrder();
     $$('.totemBlocks .totemBlock').each(function(item){
-      this.hookEditButton(item);
-      this.hookEditSourceButton(item);
+      //this.hookEditButton(item);
```
Comment (ldubost) [inline src/main/resources/TotemCode/TotemSheet.xml]: You should not commit commented code. You should remove that code.

File: src/main/resources/TotemCode/TotemSheet.xml [hunk 4]
```diff
@@ -545,7 +545,7 @@ require(['jquery', 'colpick'], function($, Colpick) {
     <name>TotemCode.TotemSheet</name>
     <number>2</number>
     <className>XWiki.JavaScriptExtension</className>
-    <guid>31a5edfd-5305-444c-b817-fc604f35a422</guid>
```
Comment (ldubost) [inline src/main/resources/TotemCode/TotemSheet.xml:548]: These changes should not be commited as there are not usefull. This is not a blocker so it can be commited as is but it would be better to avoid this next time.

File: src/main/resources/TotemCode/Translations.en.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ldubost) [inline src/main/resources/TotemCode/Translations.en.xml:1]: This change is not complete as we are missing the other translations. It could be also a separate commit for the translations. The english translations should be in the default Translation page with a defaultLanguage set to "en". The french translation should be in the french page.

File: src/main/resources/TotemCode/WebPreferences.xml [hunk 6]
```diff
@@ -0,0 +1,1262 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ldubost) [inline src/main/resources/TotemCode/WebPreferences.xml:1]: Could there not be some side effects of having a preference page for the space with a skin set. If it can be avoided that would be better

**Comment by ldubost** [comment]
This is an ok commit besides the Translations page. More comments in the code would be better.

**Comment by ldubost** [comment]
Also make sure to have one or more JIRA task ready for your commit that explains what the benefits are. It's better to commit changes unrelated to the jira task separately, especially commits for formatting or for moving code around

File: src/main/resources/TotemCode/SimpleTotemItemSheet.xml [hunk 7]
```diff
@@ -22,10 +22,137 @@
...
+    <property>
+      <cache>forbid</cache>
+    </property>
+    <property>
+      <code>document.observe("xwiki:actions:save", function(event){
+  require(['jquery'],function($){
+     var jqEvent = $(event);
+     var targetBtnName = jqEvent[0].target.activeElement.name;
+     //if( (targetBtnName == "action_saveandcontinue") || (targetBtnName == "action_upload") ){
+        console.log(targetBtnName);
```
Comment (ldubost) [inline src/main/resources/TotemCode/SimpleTotemItemSheet.xml]: This should not be staying in the commit since it can fail on some browser and log too much data

File: src/main/resources/TotemCode/TotemSheet.xml (hunk 4 above)
Comment (mouhb) [inline src/main/resources/TotemCode/TotemSheet.xml:548]: You speak about guid line ?

File: src/main/resources/TotemCode/SimpleTotemItemWebHomeSheet.xml [hunk 8]
```diff
@@ -22,10 +22,184 @@
...
+      success:function(rep, textStatus, jqXHR){
+         //$("#totemModal .modal-body").html(rep);
+         if(rep){ 
+             if(rep.results == false){
+                alert("The item name already exists!");
+             }
+             else
+             {
+                //Redirect
+                console.log(rep.results);
```
Comment (cjdelisle) [inline src/main/resources/TotemCode/SimpleTotemItemWebHomeSheet.xml]: The way I deal with console.log's is to add this at the top:

``` javascript
var debug = function () { };
//debug = function (x) { console.log(x); }; // uncomment to see debugging lines
```

then use `debug("blah blah message")`
